src/services/inlayHints.ts

```diff
@@ -296,11 +295,8 @@ export function provideInlayHints(context: InlayHintsContext): InlayHint[] {
             return;
         }
 
-        const candidates: Signature[] = [];
-        const signature = checker.getResolvedSignatureForSignatureHelp(expr, candidates);
-        if (!signature || !candidates.length) {
-            return;
-        }
+        const signature = checker.getResolvedSignature(expr);

```

Comment:
The fact that this function is enough now and doesn't break tests is pretty surprising- I think it relates to some work @Andarist might've done here?

Reply:
> The fact that this function is enough now and doesn't break tests is pretty surprising- I think it relates to some work @Andarist might've done here.

Reply:
I think it's unrelated. I was working on some related-ish stuff at different layers. It might just be that this should have never been using `getResolvedSignatureForSignatureHelp`

If we consider this test case:

```ts
function overloadTest(first: string): unknown
function overloadTest(second: string, extra: boolean): unknown
function overloadTest(...arg: any[]) {}

overloadTest('str',)
```

We get `second` as the inlay hint for our argument but the overload that is successfully resolved is the first one. I think this one should give `first` as the inlay hint and that this PR might have fixed this. It would be good to add this as a new test case.